controllers/topics/getLatestPost.js

```diff
@@ -19,58 +19,38 @@ module.exports = async (req, res) => {
       });
     }
 
-    let post_topics = await sequelize.query(
-      `SELECT
-        pt.topic_id,
-        p.*,
-        pAnonUserInfo.anon_user_info_color_code,
-        pAnonUserInfo.anon_user_info_color_name,
-        pAnonUserInfo.anon_user_info_emoji_code,
-        pAnonUserInfo.anon_user_info_emoji_name,
-        u.username,
-        u.is_anonymous,
-        u.profile_pic_path
-      FROM posts p 
-      INNER JOIN post_topics pt 
-        ON p.post_id = pt.post_id
-      INNER JOIN post_anon_user_info pAnonUserInfo
-        ON pAnonUserInfo.post_id = p.getstream_activity_id
-      INNER JOIN users u
-        ON u.user_id = p.author_user_id
-      WHERE 
-        pt.topic_id = :topic_id
-        AND p.duration >= now()
-      ORDER BY created_at DESC
-      LIMIT 1`,
-      {
-        type: QueryTypes.SELECT,
-        replacements: {
-          topic_id: topics.topic_id
-        },
-        raw: true
-      }
-    );
+    const feedClient = stream.connect(process.env.API_KEY, process.env.SECRET, process.env.APP_ID);
+    let feed = undefined;
 
-    if (post_topics?.length < 1) {
-      res.status(400).json({
-        success: false,
-        message: 'This topic has no active post'
+    try {
+      const clientFeed = feedClient.feed('topic', name, token);
+      const topicFeed = await clientFeed.get({
+        limit: 1,
+        enrich: true
       });
-    }
 
-    let post = post_topics?.at(0);
+      if (topicFeed?.results?.length < 1) {
+        return res.status(400).json({
+          success: false,
+          message: 'This topic has no active post'
+        });
+      }
 
-    let authorUsername = post?.is_anonymous
-      ? `${post?.anon_user_info_color_name} ${post?.anon_user_info_emoji_name}`
-      : post?.username;
+      feed = topicFeed?.results?.at(0);
+    } catch (e) {
+      console.error('Error getting feed', e);
+    }
+
+    let authorUsername = feed?.anonimity
+      ? `${feed?.anon_user_info_color_name} ${feed?.anon_user_info_emoji_name}`
+      : feed?.actor?.data?.username;
 
     let message = `There are new posts from ${authorUsername}`;
 
     const getstreamClient = new StreamChat(process.env.API_KEY, process.env.SECRET);
     let unreadCount = 0;
     try {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There are a few issues with this code block:

1. The `token` variable is not defined before it's used on line 26 and line 53. This will cause an error at runtime.
2. The `userId` variable is also not defined before it's used on line 53. This will cause an error at runtime.
3. Error handling for the GetStream feed retrieval is not sufficient. In case of an error, the function just logs the error and continues execution. It would be better to return a response with an error message to the client.

Here's how you can fix these issues:

```diff
    const feedClient = stream.connect(process.env.API_KEY, process.env.SECRET, process.env.APP_ID);
    let feed = undefined;
 
    try {
      const clientFeed = feedClient.feed('topic', name, token);
      const topicFeed = await clientFeed.get({
        limit: 1,
        enrich: true
       });
 
      if (topicFeed?.results?.length < 1) {
        return res.status(400).json({
          success: false,
          message: 'This topic has no active post'
        });
      }
 
      feed = topicFeed?.results?.at(0);
    } catch (e) {
      console.error('Error getting feed', e);
+42:       return res.status(500).json({
+43:         success: false,
+44:         message: 'Error retrieving feed from GetStream'
+45:       });
    }

    let authorUsername = feed?.anonimity
      ? `${feed?.anon_user_info_color_name} ${feed?.anon_user_info_emoji_name}`
      : feed?.actor?.data?.username;
 
     let message = `There are new posts from ${authorUsername}`;
 
     const getstreamClient = new StreamChat(process.env.API_KEY, process.env.SECRET);
     let unreadCount = 0;
     try {
+53:        const token = 'your_token_here'; // replace with your actual token
+54:        const userId = 'your_user_id_here'; // replace with your actual user id
       await getstreamClient.connectUser({id: userId}, token);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->